handlerset.go

```diff
@@ -3,57 +3,215 @@ package configkit
 import "github.com/dogmatiq/configkit/message"
 
 // HandlerSet is a collection of handlers.
-type HandlerSet []Handler
+type HandlerSet map[Identity]Handler
+
+// Add adds a handler to the set.
+//
+// It returns true if the handler was added, or false if the set already
+// contained a handler with the same name or key as h.
```

Comment:
What's the difference between name and key?

Reply:
Take a look at the docs of the `Identity()` method [here](https://godoc.org/github.com/dogmatiq/dogma#AggregateConfigurer).

Reply:
Cool, thanks